### PR TITLE
Jw/reddit crawler

### PR DESCRIPTION
**Pull request by jackwilliamson**

### CODE REVIEW

File: src/crawler/RedditCrawler.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            for index, post in enumerate(hot_posts):
+                if post.media is not None and 'oembed' in post.media.keys():
+                    if await self.check_new_url(post.url):
+                        title = post.media['oembed']['title']
+                        video_source = post.media['type']
+                        reddit_tag = {
+                            'id': post.id,
+                            'title': post.title
+                        }
+                        metadata = MetaDataItem("", title, post.url, video_source, "", "", "")
```
Comment (tonymeng1) [inline src/crawler/RedditCrawler.py:33]: What's the difference between id and title in MetaDataItem() and the ones in reddit_tag?

Comment (jackwilliamson) [reply src/crawler/RedditCrawler.py:33]: The ones in the reddit tag are the id and title for the post in reddit itself. The title field on the MetadataItem is the title from the video itself (like the youtube title) and the id field of MetadataItem is provided by the data system (like Firebase  or the Local Storage)

**Comment by tonymeng1** [comment]
In FirebaseAccessor.py:
```
key = ref.push(metadata.to_json()).key
metadata.id = key
```

If the id is provided by firebase, should I remove it from the constructor? It's constructed with a placeholder every time

**Comment by jackwilliamson** [comment]
Ya id is usually going to be initialized to a default value of empty unless its being pulled from the database, so I'll make that update before merging it.

**Pull request merged**